acme/http_challenge_server.go

```diff
@@ -58,7 +59,15 @@ func (s *HTTPProviderServer) serve(domain, token, keyAuth string) {
 	// For validation it then writes the token the server returned with the challenge
 	mux := http.NewServeMux()
 	mux.HandleFunc(path, func(w http.ResponseWriter, r *http.Request) {
-		if strings.HasPrefix(r.Host, domain) && r.Method == "GET" {
+		requestHost := r.Host
+		if s.allowXForwardedHost {
+			xForwardedHost := r.Header.Get("x-forwarded-host")
+			if xForwardedHost != "" {
+				requestHost = xForwardedHost
+			}
+		}
```

Comment:
Why can't this be done in pre-processing? I'd much rather this kind of thing live outside the acme lib, it seems like kind of a hack.